litmus-portal/cluster-agents/subscriber/pkg/k8s/operations.go

```diff
@@ -2,42 +2,128 @@ package k8s
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"log"
 	"os"
+	"strings"
+	"sync"
+	"time"
 
 	yaml_converter "github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	yaml2 "gopkg.in/yaml.v2"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	memory "k8s.io/client-go/discovery/cached"
 	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/restmapper"
-
-	memory "k8s.io/client-go/discovery/cached"
 )
 
 const (
 	ExternAgentConfigName = "agent-config"
+	LiveCheckMaxTries     = 6
 )
 
+type AgentComponents struct {
+	Deployments      []string `yaml:"DEPLOYMENTS"`
+	LiveStatus       bool
+	AccessLiveStatus sync.Mutex
+}
+
 var (
 	Ctx             = context.Background()
 	decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
 	dr              dynamic.ResourceInterface
 	AgentNamespace  = os.Getenv("AGENT_NAMESPACE")
 )
 
+func CheckComponentStatus() (string, error) {
+	clientset, err := GetGenericK8sClient()
+	if err != nil {
+		return "", err
+	}
+	getCM, err := clientset.CoreV1().ConfigMaps(AgentNamespace).Get(ExternAgentConfigName, metav1.GetOptions{})
```

Comment:
Instead of reading it from configmap, you use the env.

Reply:
No there can be many more components and the list can grow, also there can be other resources we might want to check currently it's just deployments. That's why I kept it in the cm

Reply:
This will be very rare, the configmap is coming from the server.

Reply:
Not exactly, when we scale the components and bring in more services it will grow, this in my opinion is a better way to handle this

Reply:
That's we move from deployment env to configmap just to read the env

Reply:
For example In Prometheus, if you add a new config to configmap, it requires a restart of the pod.

Reply:
If we follow this approch, then we don't need configmap.

Reply:
The configmap doesn't need the be mounted. We can apply and reference it from the subscriber "externally".

Reply:
Yes, @ksatchit right now we have a configmap called `agent-config` where we store all the envs.

Reply:
so, COMPONENTS env is also storing in configmap coming from the server. We can access this ENV by `os.getenv("COMPONENTS")` instead of reading by client-go again.

Reply:
With this it should auto-restart, also this deployment spec is also coming from server. It may be beneficial to keep it this way @rajdas98

Reply:
Then we don't need the configmap.

Reply:
got it, fixed.

Reply:
The configmap upgrade, the restart will be done by an upgrade agent which we are planning down the line.

Reply:
Then it's good with the cm